pkgs/development/misc/resholve/oildev.nix

```diff
@@ -0,0 +1,121 @@
+{ lib
+, stdenv
+, python27Packages
+, callPackage
+, fetchFromGitHub
+, makeWrapper
+, # re2c deps
+  autoreconfHook
+, # py-yajl deps
+  git
+, # oil deps
+  readline
+, cmark
+, file
+, glibcLocales
+}:
+
+rec {
+  re2c = stdenv.mkDerivation rec {
+    pname = "re2c";
+    version = "1.0.3";
+    sourceRoot = "${src.name}/re2c";
+    src = fetchFromGitHub {
+      owner = "skvadrik";
+      repo = "re2c";
+      rev = version;
+      sha256 = "0grx7nl9fwcn880v5ssjljhcb9c5p2a6xpwil7zxpmv0rwnr3yqi";
+    };
+    nativeBuildInputs = [ autoreconfHook ];
+    preCheck = ''
+      patchShebangs run_tests.sh
+    '';
+  };
+
+  py-yajl = python27Packages.buildPythonPackage rec {
+    pname = "oil-pyyajl-unstable";
+    version = "2019-12-05";
+    src = fetchFromGitHub {
+      owner = "oilshell";
+      repo = "py-yajl";
+      rev = "eb561e9aea6e88095d66abcc3990f2ee1f5339df";
+      sha256 = "17hcgb7r7cy8r1pwbdh8di0nvykdswlqj73c85k6z8m0filj3hbh";
+      fetchSubmodules = true;
+    };
+    # just for submodule IIRC
+    nativeBuildInputs = [ git ];
+  };
+
+  oildev = python27Packages.buildPythonPackage rec {
+    pname = "oildev-unstable";
+    version = "2021-07-14";
+
+    src = fetchFromGitHub {
+      owner = "oilshell";
+      repo = "oil";
+      # rev == present HEAD of release/0.8.12
+      rev = "799c0703d1da86cb80d1f5b163edf9369ad77cf1";
+      hash = "sha256-QNSISr719ycZ1Z0quxHWzCb3IvHGj9TpogaYz20hDM4=";
+
+      /*
+        It's not critical to drop most of these; the primary target is
+        the vendored fork of Python-2.7.13, which is ~ 55M and over 3200
+        files, dozens of which get interpreter script patches in fixup.
+      */
+      extraPostFetch = ''
+        rm -rf Python-2.7.13 benchmarks metrics py-yajl rfc gold web testdata services demo devtools cpp
```

Comment:
This is a case where not forcing is a _real_ chore. The command fails if the files are missing, which *will* be the case when you have updated the rev and need a new hash for TOFU. i.e.:
```console
$ nix-build -A resholve
Wed Sep 22 2021 09:10:35 --> 
warning: warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
these 3 derivations will be built:
  /nix/store/drmwnyrlsbymisniys6qvsbmhshrilw1-source.drv
  /nix/store/pm05pp5494z1sihri14imhirqgskzqjf-python2.7-oildev-unstable-2021-07-14.drv
  /nix/store/sq5ag7kwcqnrymfjxfncwg7vf8vpwdzq-resholve-0.6.0.drv
building '/nix/store/drmwnyrlsbymisniys6qvsbmhshrilw1-source.drv'...

trying https://github.com/oilshell/oil/archive/df9f73af41d4f9a4963a7a146965d223feacf484.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   154  100   154    0     0    234      0 --:--:-- --:--:-- --:--:--   233
100 14.7M    0 14.7M    0     0  2605k      0 --:--:--  0:00:05 --:--:-- 3274k
unpacking source archive /private/tmp/nix-build-source.drv-0/df9f73af41d4f9a4963a7a146965d223feacf484.tar.gz
rm: cannot remove 'Python-2.7.13': No such file or directory
rm: cannot remove 'benchmarks': No such file or directory
rm: cannot remove 'metrics': No such file or directory
rm: cannot remove 'py-yajl': No such file or directory
rm: cannot remove 'rfc': No such file or directory
rm: cannot remove 'gold': No such file or directory
rm: cannot remove 'web': No such file or directory
rm: cannot remove 'testdata': No such file or directory
rm: cannot remove 'services': No such file or directory
rm: cannot remove 'demo': No such file or directory
rm: cannot remove 'devtools': No such file or directory
rm: cannot remove 'cpp': No such file or directory
error: builder for '/nix/store/drmwnyrlsbymisniys6qvsbmhshrilw1-source.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/pm05pp5494z1sihri14imhirqgskzqjf-python2.7-oildev-unstable-2021-07-14.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sq5ag7kwcqnrymfjxfncwg7vf8vpwdzq-resholve-0.6.0.drv' failed to build
```

instead of 

```console
$ nix-build -A resholve
Wed Sep 22 2021 09:11:10 --> 
warning: warning: Nix search path entry '/nix/var/nix/profiles/per-user/root/channels' does not exist, ignoring
these 3 derivations will be built:
  /nix/store/6hqh0ggmydl03smw2zsk455gdqdc35ml-source.drv
  /nix/store/9qbyjv40aakhfdgdh0agz304ynsl7bkc-python2.7-oildev-unstable-2021-07-14.drv
  /nix/store/csc8jwpkg2j00wm3v6hh9vw7h2kz97a6-resholve-0.6.0.drv
building '/nix/store/6hqh0ggmydl03smw2zsk455gdqdc35ml-source.drv'...

trying https://github.com/oilshell/oil/archive/df9f73af41d4f9a4963a7a146965d223feacf484.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   154  100   154    0     0    492      0 --:--:-- --:--:-- --:--:--   492
100 14.7M    0 14.7M    0     0  3534k      0 --:--:--  0:00:04 --:--:-- 4203k
unpacking source archive /private/tmp/nix-build-source.drv-0/df9f73af41d4f9a4963a7a146965d223feacf484.tar.gz
error: hash mismatch in fixed-output derivation '/nix/store/6hqh0ggmydl03smw2zsk455gdqdc35ml-source.drv':
         specified: sha256-0Ej7HCmtHZWmGrshIlgA9V5DSR4mdsQ+XPYmWZUOGoo=
            got:    sha256-hMyoSuXqm+dHemZKibDHaOVbeb4iiIDU29SX3itK7tw=
error: 1 dependencies of derivation '/nix/store/9qbyjv40aakhfdgdh0agz304ynsl7bkc-python2.7-oildev-unstable-2021-07-14.drv' failed to build
error: 1 dependencies of derivation '/nix/store/csc8jwpkg2j00wm3v6hh9vw7h2kz97a6-resholve-0.6.0.drv' failed to build
```

I've documented this in https://github.com/abathur/nix-py-dev-oil/commit/54ffc4bc666680e2afbbed41bce6bea6b6113fdb